spec/core/v2/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -784,14 +819,45 @@ function writeAcknowledgement(
     // Note that the event should be emitted by this function only in the asynchrounous ack case. Otherwise the event is emitted during the onReceive 
     packet=getPacket(destChannelId,sequence)
     if(packet!=nil){
-        emitLogEntry("writeAcknowledgement", {
-        sequence: packet.sequence,
-        sourceId: packet.channelSourceId,
-        destId: packet.channelDestId,
-        timeoutTimestamp: packet.timeoutTimestamp,
-        data: packet.data,
-        acknowledgement
+        // emit an acknowledgement event for each app acknowledgement in the acknowledgement
+        for i, ack in acknowledgment {
+            emitEvents("write_acknowledgement", {
+                sourceChannel: packet.sourceChannel,
+                destChannel: packet.destChannel,
+                sequence: packet.sequence, // value is string in decimal format
+                payloadSequence: i, // value is string in decimal format
+                acknowledgement: toHex(ack),
+            })
+        }
+
+        // Event Emission for receive packet. emit again so relayer can reconstruct the packet
```

Comment:
Note that re-emission may be expensive. Can't relayers store somewhere these events instead?